clients/gateway/integration_test.go

```diff
@@ -0,0 +1,92 @@
+//go:build integration
+// +build integration
+
+package gateway
```

Comment:
We put our tests in `*_test` packages
```suggestion
package gateway_test
```